### PR TITLE
GH2874: Merge Frosting into Cake

### LINKED ISSUE #2874: Merge frosting into main Cake repo
We should consider meging Frosting into main https://github.com/cake-build/cake repository. They both share the same release lifecycle and having them in a single repository would simplify and allow more automation in release process.

### PR DESCRIPTION
**Pull request by devlead**
This is WIP to see what it would entail to fix #2874 

Changes
* [x] src + template from frosting, preserving contributor history
* [x] Filters out example from Cake build script pack
* [x] Adds Frosting projects to Cake.sln
* [x] Adds template frosting version poke for CI builds
* [x] Adds net5.0 moniker for lib & tests
* [x] Fix so project follows Cake style guidelines
* [x] Fix Cake 1.0 breaking changes
* [x] Move frosting bootstappers to template
  * [x] set execute bit build.sh
* [x] Adjusted all issue references to cake-build/frosting
* [x] Integration tests
* [x] Adjust .NET Template author

### CODE REVIEW

**Comment by devlead** [comment]
@cake-build/cake-team this should now be ready for review.

File: src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "author": "Patrik Svensson (patrik@patriksvensson.se)",
```
Comment (pascalberger) [inline src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json:2]: Do we want to keep @patriksvensson  as author here or switch to same as for the other runners?

**Review by pascalberger — COMMENTED** [review]
I've added one question as comment.

What's the reason for the failing build on Windows?

File: src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json (hunk 1 above)
Comment (patriksvensson) [reply src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json:2]: Switch to the same as other runners.

Comment (devlead) [reply src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json:2]: This is in the template manifest not nuget packages. But can investigate charging this.

**Comment by devlead** [comment]
> What's the reason for the failing build on Windows?
Believe it's a split overload issue, pushed commit to see

**Comment by devlead** [comment]
> > What's the reason for the failing build on Windows?
> 
> Believe it's a split overload issue, pushed commit to see

Yes it was, all green now.

File: src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json (hunk 1 above)
Comment (devlead) [reply src/Cake.Frosting.Template/templates/cakefrosting/.template.config/template.json]: Changed to the same as NuGet packages.
Doesn't look like there're any special rules around this property, so probably just works.
image

**Review by pascalberger — APPROVED** [review]
LGTM

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@devlead tha is for getting this done!

**Pull request merged**